framework/src/org/checkerframework/common/value/Range.java

```diff
@@ -0,0 +1,742 @@
+        if (this.isNothing() || right.isNothing()) {
+            return NOTHING;
+        }
+
+        if (right.from == Long.MAX_VALUE) {
+            return NOTHING;
+        }
+
+        long resultFrom = Math.max(from, right.from + 1);
+        return createRangeOrNothing(resultFrom, to);
+    }
+
+    /**
+     * Refines this range to reflect that some value in it can be greater than or equal to a value
+     * in the given range. This is used for calculating the control-flow-refined result of the &gt;=
+     * operator. For example:
+     *
+     * <pre>
+     * <code>
+     *     {@literal @}IntRange(from = 0, to = 10) int a;
+     *     {@literal @}IntRange(from = 3, to = 7) int b;
+     *     ...
+     *     if (a &gt;= b) {
+     *         // range of <i>a</i> is now refined to [3, 10] because a value in range [0, 2]
+     *         // cannot be greater than or equal to variable <i>b</i> with range [3, 7].
+     *         ...
+     *     }
+     * </code>
+     * </pre>
+     *
+     * Use the {@link #refineLessThan(Range)} method if you are also interested in refining the
+     * given range
+     *
+     * @param right the specified {@code Range} to compare with
+     * @return the refined {@code Range}
+     */
+    public Range refineGreaterThanEq(Range right) {
+        if (this.isNothing() || right.isNothing()) {
+            return NOTHING;
+        }
+
+        long resultFrom = Math.max(from, right.from);
+        return createRangeOrNothing(resultFrom, to);
+    }
+
+    /**
+     * Refines this range to reflect that some value in it can be equal to a value in the given
+     * range. This is used for calculating the control-flow-refined result of the == operator. For
+     * example:
+     *
+     * <pre>
+     * <code>
+     *     {@literal @}IntRange(from = 0, to = 10) int a;
+     *     {@literal @}IntRange(from = 3, to = 15) int b;
+     *     ...
+     *     if (a == b) {
+     *         // range of <i>a</i> is now refined to [3, 10] because a value in range [0, 2]
+     *         // cannot be equal to variable <i>b</i> with range [3, 15].
+     *         ...
+     *     }
+     * </code>
+     * </pre>
+     *
+     * @param right the specified {@code Range} to compare with
+     * @return the refined {@code Range}
+     */
+    public Range refineEqualTo(Range right) {
+        if (this.isNothing() || right.isNothing()) {
+            return NOTHING;
+        }
+
+        long resultFrom = Math.max(from, right.from);
+        long resultTo = Math.min(to, right.to);
+        return createRangeOrNothing(resultFrom, resultTo);
+    }
+
+    /**
+     * Determines if the range is wider than a given value, i.e., if the number of possible values
+     * enclosed by this range is more than the given value.
+     *
+     * @param value the value to compare with
+     * @return true if wider than the given value
+     */
+    public boolean isWiderThan(long value) {
+        return BigInteger.valueOf(to)
```

Comment:
`isWiderThan` could use more efficient primitive operations if it checked whether `from` and `to` are within Integer widths, as other methods do.